### ISSUE TITLE
Avalonia app build failed 

### ISSUE THREAD

**Issue by timiil** [dotnet/runtimelab#926]
hello all, here is my steps:

* 1, create a container,
```
docker run -it --name=dntest mcr.microsoft.com/dotnet/sdk:5.0 /bin/bash

```

* 2,  inside the docker,

```
apt-get update -y && apt-get upgrade -y && apt-get install -y nano clang zlib1g-dev libkrb5-dev

cd /home

dotnet new -i Avalonia.Templates

dotnet new avalonia.app -o MyApp

cd MyApp

dotnet new nuget

```

* 4, edit the nuget.config
```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
    <clear />
    <add key="nuget" value="https://api.nuget.org/v3/index.json" />
    <add key="exp" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-experimental/nuget/v3/index.json" />
</packageSources>
</configuration>
```

* 5, 
```
dotnet add package Microsoft.DotNet.ILCompiler --prerelease

dotnet publish ./MyApp.csproj -c Release -r linux-x64
```

* 6, after serveal miniutes and tons of warrnings, the error come:

```
EXEC : error : Value cannot be null. (Parameter 'key') [/home/MyApp/MyApp.csproj]
  System.ArgumentNullException: Value cannot be null. (Parameter 'key')
     at System.Collections.Generic.Dictionary`2.FindValue(TKey key)
     at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
     at ILCompiler.AnalysisBasedMetadataManager.Policy.GeneratesMetadata(EcmaModule module, ExportedTypeHandle exportedTypeHandle)
     at ILCompiler.Metadata.Transform`1.HandleTypeForwarders(EcmaModule module)
     at ILCompiler.Metadata.Transform`1.InitializeScopeDefinition(ModuleDesc module, ScopeDefinition scopeDefinition)
     at ILCompiler.Metadata.EntityMap`2.GetOrCreate[TConcreteEntity,TConcreteRecord](TConcreteEntity entity, Action`2 initializer)
     at ILCompiler.Metadata.Transform`1.HandleNamespaceDefinition(ModuleDesc parentScope, String namespaceString)
     at ILCompiler.Metadata.Transform`1.InitializeTypeDef(MetadataType entity, TypeDefinition record)
     at ILCompiler.Metadata.EntityMap`2.Create[TConcreteEntity,TConcreteRecord](TConcreteEntity entity, Action`2 initializer)
     at ILCompiler.Metadata.Transform`1.HandleType(TypeDesc type)
     at ILCompiler.Metadata.MetadataTransform.Run[TPolicy](TPolicy policy, IEnumerable`1 modules)
     at ILCompiler.GeneratingMetadataManager.ComputeMetadata[TPolicy](TPolicy policy, NodeFactory factory, Byte[]& metadataBlob, List`1& typeMappings, List`1& methodMappings, List`1& fieldMappings, List`1& stackTraceMapping)
     at ILCompiler.AnalysisBasedMetadataManager.ComputeMetadata(NodeFactory factory, Byte[]& metadataBlob, List`1& typeMappings, List`1& methodMappings, List`1& fieldMappings, List`1& stackTraceMapping)
     at ILCompiler.DependencyAnalysis.MetadataNode.GetData(NodeFactory factory, Boolean relocsOnly)
     at ILCompiler.DependencyAnalysis.ObjectWriter.EmitObject(String objectFilePath, IEnumerable`1 nodes, NodeFactory factory, ObjectWritingOptions options, IObjectDumper dumper)
     at ILCompiler.Compilation.ILCompiler.ICompilation.Compile(String outputFile, ObjectDumper dumper)
     at ILCompiler.Program.Run(String[] args)
     at ILCompiler.Program.Main(String[] args)
/root/.nuget/packages/microsoft.dotnet.ilcompiler/6.0.0-preview.4.21206.3/build/Microsoft.NETCore.Native.targets(259,5): error MSB3073: The command ""/root/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/6.0.0-preview.4.21206.3/tools/ilc" @"obj/Release/net5.0/linux-x64/native/MyApp.ilc.rsp"" exited with code 1. [/home/MyApp/MyApp.csproj]

```

any help is very wellcome, thanks!

**Comment by MichalStrehovsky** [comment]
https://github.com/dotnet/runtime/pull/50845 should fix this. It might take a bit until it flows here.

If you can use a ILCompiler package from about a week ago, that would also fix this since it doesn't have the problematic codepath.

**Comment by timiil** [comment]
> 
> 
> [dotnet/runtime#50845](https://github.com/dotnet/runtime/pull/50845) should fix this. It might take a bit until it flows here.
> 
> If you can use a ILCompiler package from about a week ago, that would also fix this since it doesn't have the problematic codepath.

thanks , i had pick up an early package that is OK for now.

**Comment by MichalStrehovsky** [comment]
Fixed in #936
